datahub-web-react/src/app/lineage/utils/extendAsyncEntities.ts

```diff
@@ -42,24 +42,78 @@ function updateFineGrainedMap(
     mapForFieldReverse[upstreamEntityUrn] = listForDownstreamReverse;
 }
 
-function extendColumnLineage(lineageVizConfig: FetchedEntity, fineGrainedMap: any) {
+function extendColumnLineage(
+    lineageVizConfig: FetchedEntity,
+    fineGrainedMap: any,
+    fineGrainedMapForSiblings: any,
+    fetchedEntities: FetchedEntities,
+) {
     if (lineageVizConfig.fineGrainedLineages && lineageVizConfig.fineGrainedLineages.length > 0) {
         lineageVizConfig.fineGrainedLineages.forEach((fineGrainedLineage) => {
             fineGrainedLineage.upstreams?.forEach((upstream) => {
                 const [upstreamEntityUrn, upstreamField] = breakFieldUrn(upstream);
                 fineGrainedLineage.downstreams?.forEach((downstream) => {
-                    const [downstreamEntityUrn, downstreamField] = breakFieldUrn(downstream);
+                    const downstreamField = breakFieldUrn(downstream)[1];
+                    // fineGrainedLineage always belongs on the downstream urn with upstreams pointing to another entity
+                    // pass in the visualized node's urn and not the urn from the schema field as the downstream urn,
+                    // as they will either be the same or if they are different, it belongs to a "hidden" sibling
                     updateFineGrainedMap(
                         fineGrainedMap,
                         upstreamEntityUrn,
                         upstreamField,
-                        downstreamEntityUrn,
+                        lineageVizConfig.urn,
                         downstreamField,
                     );
+
+                    // upstreamEntityUrn could belong to a sibling we don't "render", so store its inputs to updateFineGrainedMap
+                    // and update the fine grained map later when we see the entity with these siblings
+                    // eslint-disable-next-line no-param-reassign
```

Comment:
this is a bit challenging for me to grasp...

Reply:
Okay got it. Yeah makes sense - seems like siblings + graph is a really tricky combo. As long as this doesn't introduce any regression I'm happy! :)

Reply:
haha for sure! things are looking good locally for me but i'll make sure to give it a once more around before merging.